lib/iris/fileformats/cf.py

```diff
@@ -1416,16 +1419,66 @@ def _translate(self, variables):
         # Identify and register all CF formula terms.
         formula_terms = _CFFormulaTermsVariable.identify(variables)
 
-        for cf_var in formula_terms.values():
-            for cf_root, cf_term in cf_var.cf_terms_by_root.items():
-                # Ignore formula terms owned by a bounds variable.
-                if cf_root not in self.cf_group.bounds:
-                    cf_name = cf_var.cf_name
-                    if cf_var.cf_name not in self.cf_group:
-                        self.cf_group[cf_name] = CFAuxiliaryCoordinateVariable(
-                            cf_name, cf_var.cf_data
-                        )
-                    self.cf_group[cf_name].add_formula_term(cf_root, cf_term)
+        if iris.FUTURE.derived_bounds:
+            # cf_var = CFFormulaTermsVariable (loops through everything that appears in formula terms)
+            for cf_var in formula_terms.values():
+                # eg. eta:'a' | cf_root = eta and cf_term = a. cf_var.cf_terms_by_root = {'eta': 'a'} (looking at all appearances in formula terms)
+                for cf_root, cf_term in cf_var.cf_terms_by_root.items():
+                    # gets set to the bounds of the coord from cf_root_coord
+                    bounds_name = None
+                    # cf_root_coord = CFCoordinateVariable of the coordinate relating to the root
+                    cf_root_coord = self.cf_group.coordinates.get(cf_root)
+                    if cf_root_coord is None:
+                        cf_root_coord = self.cf_group.auxiliary_coordinates.get(cf_root)
+                    with contextlib.suppress(AttributeError):
+                        # Copes with cf_root_coord not existing, OR not having
+                        #  `bounds` attribute.
+                        bounds_name = cf_root_coord.bounds
+
+                    if bounds_name is not None:
+                        try:
```

Comment:
Ultimately, I think you'll probably want to replace this try-except call. I expect you ought to be able to replace this with something like:

```python
bounds_vars = [...]
if len(bounds_vars) == 1:
  bounds_var = bounds_vars[0]
  ...
else:
  ...
```